sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2178,6 +2209,11 @@ def retry_eeprom_reading(self):
         # Update retry EEPROM set
         self.retry_eeprom_set -= retry_success_set
 
+        # If we were here and now retry_eeprom_set is empty, then we can state
+        # that all SFPs accessible
+        if not self.retry_eeprom_set:
+            notify_system_ready()
```

Comment:
Why do you correlate ready of syseeprom with pmon ready? There are many other initializations done in pmon like sfp, thermal etc. Shouldn't you set pmon status ready only after all these tasks are done?